From 4212aa4087c3e2c129c29d3cc62c530559328c79 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Fri, 13 Nov 2020 10:37:18 +0800
Subject: [PATCH] video: rockchip: mpp: rkvenc issue for devfreq is null

CONFIG_PM_DEVFREQ enable, while devfreq is not set in dtsi.
then freq is not actually set.

Change-Id: Ie615d38393155c0c0debdc4522f1d47869ea15a2
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_rkvenc.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/mpp/mpp_rkvenc.c b/drivers/video/rockchip/mpp/mpp_rkvenc.c
index f0e097a0f0d2..21b08e91df19 100644
--- a/drivers/video/rockchip/mpp/mpp_rkvenc.c
+++ b/drivers/video/rockchip/mpp/mpp_rkvenc.c
@@ -1319,9 +1319,8 @@ static int rkvenc_set_freq(struct mpp_dev *mpp,
 		mutex_unlock(&enc->devfreq->lock);
 		return 0;
 	}
-#else
-	mpp_clk_set_rate(&enc->core_clk_info, task->clk_mode);
 #endif
+	mpp_clk_set_rate(&enc->core_clk_info, task->clk_mode);
 
 	return 0;
 }
-- 
2.35.3

